ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/BaseOptimizeCommit.java

```diff
@@ -422,4 +425,31 @@ private static Set<ContentFile<?>> selectMajorOptimizeDeletedFiles(BaseOptimizeT
 
     return result;
   }
+
+  private static Set<String> getCommittedDataFilesFromSnapshotId(ArcticTable arcticTable, long snapshotId) {
+    UnkeyedTable baseArcticTable;
+    if (arcticTable.isKeyedTable()) {
+      baseArcticTable = arcticTable.asKeyedTable().baseTable();
+    } else {
+      baseArcticTable = arcticTable.asUnkeyedTable();
+    }
+
+    long currentSnapshotId = UnKeyedTableUtil.getSnapshotId(baseArcticTable);
+    if (currentSnapshotId == snapshotId) {
```

Comment:
The following `SnapshotUtil.ancestorsBetween` has handled this case, so it could be removed.

Reply:
And what if `snapshotId == TableOptimizeRuntime.INVALID_SNAPSHOT_ID`, which will happen after planning on an empty base table?

Maybe we should change `snapshotId` from `TableOptimizeRuntime.INVALID_SNAPSHOT_ID` to `null` before we call `SnapshotUtil.ancestorsBetween`.

Reply:
OK